src/pages/signup.js

```diff
@@ -0,0 +1,260 @@
+
+import gear from "../assets/gear.svg"
+import PasswordStrengthMeter from "../components/passwordstrengthmeter"
+
+import { faGoogle, faGithub } from "@fortawesome/free-brands-svg-icons"
+import {auth,
+  signInWithGoogle,
+  signInWithGitHub,
+  prepareUserInformation,
+  getCurrentUser,
+  verifyEmail} from "../firebase/firebase"
+
+import { HeartwoodStateContext, HeartwoodDispatchContext } from "../state/HeartwoodContextProvider"
+
+library.add(faGoogle, faGithub)
+config.autoAddCss = false
+
+const SignUp = () => {
+  const state = useContext(HeartwoodStateContext)
+  const dispatch = useContext(HeartwoodDispatchContext)
+
+  const [form, setForm] = useState({email: "",
+    password: "",
+    message: { text: "", type: "" },
+    passwordStrong: false,
+    isLoading: true,
+    onVerifyLink: false})
+
+  // Generate a new document for a new user
+  const sendUserVerificationEmail = (event) => {
+    setForm({ ...form, isLoading: true })
+    event.preventDefault()
+    try {
+      // Send the verification email and tell the user to check their inbox
+      verifyEmail(form.email, "signup")
+      setForm({ ...form, message: { text: "Please check your email to complete your signup", type: "success" } })
+    } catch (error) {
+      setForm({ ...form, isLoading: false })
+
+      // handle and display the various errors to the user
+      if (error.code === "auth/account-exists-with-different-credential") {
+        setForm({ ...form, message: { text: "An account already exists under that email address", type: "error" } })
+      } else if (error.code === "auth/email-already-in-use") {
+        setForm({ ...form, message: { text: "Email address is already in use by another account", type: "error" } })
+      } else if (error.code === "auth/invalid-email") {
+        setForm({ ...form, message: { text: "Email address is badly formatted", type: "error" } })
+      } else {
+        setForm({ ...form, message: { text: "Error signing up with email and password", type: "error" } })
+      }
+    }
+  }
+
+  const onChangeHandler = (event) => {
+    const { name, value } = event.currentTarget
+    if (name === "userEmail") {
+      setForm({ ...form, email: value })
+    } else if (name === "userPassword") {
+      setForm({ ...form, password: value })
+    }
+  }
+
+  const checkStrong = (input) => {
+    if (input === "Strong") {
+      setForm({ ...form, passwordStrong: true })
+      return
+    }
+    setForm({ ...form, passwordStrong: false })
+  }
+
+  // checks that all user inputs are not empty
+  const validateInputs = () => {
+    if (!form.passwordStrong) {
+      setForm({ ...form, message: { text: "Please select a more complex password", type: "error" } })
+      return false
+    }
+    if (form.email === "" || form.password === "") {
+      setForm({ ...form, message: { text: "Error signing up with email and password", type: "error" } })
+      return false
+    }
+    return true
+  }
+
+  const submitForm = (event) => {
```

Comment:
Codacy found an issue: ['submitForm' is assigned a value but never used.](https://app.codacy.com/gh/treetoplearning/heartwood/pullRequest?prid=5919976)